rules.d/60-persistent-storage.rules.in

```diff
@@ -55,7 +55,11 @@ ENV{ID_MODEL}=="?*", ENV{ID_SERIAL_SHORT}=="?*", ENV{ID_NSID}=="?*", OPTIONS="st
 LABEL="nvme_end"
 
 # virtio-blk
+KERNEL=="vd*", ATTRS{serial}=="?*", GOTO="vdserial"
+KERNEL=="vd*", IMPORT{program}="/usr/bin/udevadm info --query=property /dev/%k", SYMLINK+="disk/by-id/virtio-$env{ID_PART_TABLE_UUID}$env{.PART_SUFFIX}", GOTO="vdend"
```

Comment:
hmm, no, this looks wrong in many levels. we don't shell out to udevadm, we *are* udev after all. but in particular "by-id" should never use part table identifiers, we already have other symlinks that cover that.

Reply:
Using `udevadm` was the only way to get a reasonable identifier. For example referencing `$env{ID_PART_ENTRY_UUID}` did not work. And it seemed better than the device index because it is robust to any changes in device ordering. Open to suggestions.